notebooks/cookbook/Enhanced sampling methods/Umbrella sampling MD.ipynb

```diff
@@ -0,0 +1,436 @@
+   "source": [
+    "# define a function for creating RMSD restraints \n",
+    "def create_rmsd_restraint(positions, atom_indicies):\n",
+    "    rmsd_cv = RMSDForce(positions, atom_indicies)\n",
+    "    energy_expression = 'step(dRMSD) * (K_RMSD/2) * dRMSD^2; dRMSD = (RMSD-RMSD0);'                                                  \n",
+    "    energy_expression += 'K_RMSD = %f;' % spring_constant.value_in_unit_system(md_unit_system)                                       \n",
+    "    energy_expression += 'RMSD0 = %f;' % restraint_distance.value_in_unit_system(md_unit_system)                                     \n",
+    "    restraint_force = CustomCVForce(energy_expression)                                                                               \n",
+    "    restraint_force.addCollectiveVariable('RMSD', rmsd_cv)                                                                           \n",
+    "    return restraint_force                                                                                                           \n",
+    "                                                                  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define a function to list force groups and modify system context                                                       \n",
+    "def forcegroupify(system):                                        \n",
+    "    forcegroups = {}                                              \n",
+    "    for i in range(system.getNumForces()):                                                                                           \n",
+    "        force = system.getForce(i)                           \n",
+    "        force.setForceGroup(i) \n",
+    "        forcegroups[force] = i                                    \n",
+    "    return forcegroups"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create system + CV Forces with harmonic restraints"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim_temp = 300.0 * kelvin\n",
+    "H_mass = 4.0 * amu #Might need to be tuned to 3.5 amu \n",
+    "time_step = 0.002 * picosecond  \n",
+    "nb_cutoff = 10.0 * angstrom                                                                                                          \n",
+    "box_padding = 12.0 * angstrom\n",
+    "salt_conc = 0.15 * molar\n",
+    "receptor_path=\"../villin.pdb\"\n",
+    "current_file=\"villin-solvated\"\n",
+    "# Misc parameters                                                 \n",
+    "restraint_distance = 0.0 * angstroms                              \n",
+    "restart_freq = 10\n",
+    "log_freq = 1                                                                                                                         \n",
+    "prd_steps = 100     "
```

Comment:
Many of these parameters are never used.  Most of the others are only used once, so you can just put the value at the appropriate place in the code without defining a variable for it.  Remember, the goal of a cookbook entry is to show how to do one specific thing with the smallest amount of code possible.  It doesn't need to be a complete, robust simulation code.